cmd/jaeger/internal/integration/integration.go

```diff
@@ -130,3 +129,63 @@ func (s *StorageIntegration) Test(t *testing.T) {
 
 	tc.ValidateData()
 }
+
+func (s *StorageIntegration) UnitTest(t *testing.T) {
+	if os.Getenv("STORAGE") != s.Name {
+		t.Skipf("Integration test against Jaeger-V2 %[1]s skipped; set STORAGE env var to %[1]s to run this", s.Name)
+	}
+	factories, err := internal.Components()
+	require.NoError(t, err)
+
+	storageCfg, exporterCfg, telemetrySettings := s.initTelemetry(t, factories)
+
+	extensionFactory := jaegerstorage.NewFactory()
+	storageExtension, err := extensionFactory.CreateExtension(
+		context.Background(),
+		extension.CreateSettings{
+			ID:                jaegerstorage.ID,
+			TelemetrySettings: telemetrySettings,
+		},
+		storageCfg,
+	)
+	require.NoError(t, err)
+
+	s.storageExtension = storageExtension
+	s.exporterCfg = exporterCfg
+
+	s.tests = &integration.StorageIntegration{}
+
+	s.initComponents(t)
+	s.tests.RunAll(t)
+}
+
+func (s *StorageIntegration) initComponents(t *testing.T) {
+	host := storageHost{t: t, storageExtension: s.storageExtension}
+	err := s.storageExtension.Start(context.Background(), host)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, s.storageExtension.Shutdown(context.Background())) })
+	storageFactory, err := jaegerstorage.GetStorageFactory(s.exporterCfg.TraceStorage, host)
+	require.NoError(t, err)
+	spanReader, err := storageFactory.CreateSpanReader()
+	require.NoError(t, err)
+	s.tests.SpanReader = spanReader
+	spanWriter, err := storageFactory.CreateSpanWriter()
+	require.NoError(t, err)
+	s.tests.SpanWriter = spanWriter
+	archiveSpanReader, err := storageFactory.CreateSpanReader()
+	require.NoError(t, err)
+	s.tests.ArchiveSpanReader = archiveSpanReader
+	archiveSpanWriter, err := storageFactory.CreateSpanWriter()
+	require.NoError(t, err)
+	s.tests.ArchiveSpanWriter = archiveSpanWriter
+
+	s.tests.Refresh = func(_ *testing.T) {}
+	s.tests.CleanUp = s.cleanUp
+}
+
+func (s *StorageIntegration) cleanUp(t *testing.T) {
+	require.NoError(t, s.storageExtension.Shutdown(context.Background()))
+
+	time.Sleep(20 * time.Second)
+	s.initComponents(t)
+}
```

Comment:
i think the issue is with the cleanup process. In the existing integration test, we simply close the factory and initialize it again. However, here, we don't have anything similar to close factory, so to tackle this, I tried the following approach -> in cleanup function
- i shutted down the storage extension and then restarted it again. got this error,
```bash
Error:          Received unexpected error:
                duplicate grpc storage name external-storage

plugin error: rpc error: code = Canceled desc = grpc: the client connection is closing

```